lib/fauna/objects.rb

```diff
@@ -60,6 +63,40 @@ def ==(other)
     alias_method :eql?, :==
   end
 
+  class Native
```

Comment:
How do we feel about meta programming? Classes like this could be much shorter with little affect in its runtime cost:

```ruby
Ref = Struct.new(:id)

class Native
  @@natives = %w(classes indexes databases functions keys tokens credentials).freeze

  @@natives.each do |id|
    instance_variable_set "@#{id}", Ref.new(id).freeze
    self.class.send :attr_reader, id.to_sym
  end

  def self.from_native(id)
    return Ref.new(id) unless @@natives.include? id
    send id.to_sym
  end
end

puts Native.keys
puts Native.from_native("keys")
puts Native.from_native("classes/users/1")
```

I think @ashfire908 should get involved in this discussion.

Regardless of the usage of meta programming, this constants must be frozen.

Reply:
Either way is fine by me (I'm down for meta programming here), but I agree that the constants should be frozen.